torch/csrc/jit/tensorexpr/stmt.h

```diff
@@ -655,6 +655,31 @@ class TORCH_API For : public StmtNode<For> {
     return new For(var_, start_, stop_, body, loop_options_);
   }
 
+  // Returns the For stmt that is immediately enclosing the given stmt.
+  static For* getParentLoop(const Stmt* st) {
+    if (st == nullptr) {
+      return nullptr;
+    }
+    auto par = st->get_parent();
+    if (auto f = dynamic_cast<For*>(par)) {
+      return f;
+    }
+    return For::getParentLoop(par);
+  }
+
+  // Returns the list of For stmts corresponding to the loopnest that is
+  // enclosing the given stmt.
+  static std::vector<For*> getEnclosingLoopNest(const Stmt* st) {
+    std::vector<For*> loops;
+    auto f = For::getParentLoop(st);
+    while (f) {
+      loops.push_back(f);
+      f = For::getParentLoop(f);
+    }
+    std::reverse(loops.begin(), loops.end());
+    return loops;
+  }
+
```

Comment:
Is there any compelling reason to make these functions static methods of `For`?  I think they should be static free functions inside loopnest.cpp instead, to minimize clutter in the API.

Reply:
I can move them to `LoopNest`, but why do you want them to be non-static? There is no use of any state here. 

@ZolotukhinM and I have been leaning towards static methods in the hope that we can get rid of the state in `LoopNest` eventually. That would make the code more clear IMO.

Reply:
Moved them to `LoopNest` but retained as `static`. Let me know if you have any reservations with that.

Reply:
Sorry , `static` is overloaded.  I meant a `static` free function (not part of a class), as opposed to a class member.  But a class member is fine too if they should be exposed to the API.

Reply:
Aah okay. Ideally, we should have them as standalone function. That's probably my end goal as well, once we make all these transforms as `static` in `LoopNest`. For now, since we have several other `static` functions (and no standalone functions), lets keep it that way for uniformity.